src/components/ProjectSettingsTabs/CreateProjectModal.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Size, Variant } from "@leafygreen-ui/button";
+import { Button } from "components/Button";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { SpruceForm, SpruceFormProps } from "components/SpruceForm";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  CreateProjectMutation,
+  CreateProjectMutationVariables,
+} from "gql/generated/types";
+import { CREATE_PROJECT } from "gql/mutations";
+
+const getModalFormDefinition = (owner: string, repo: string) => ({
+  schema: {
+    type: "object" as "object",
+    properties: {
+      projectName: {
+        type: "string" as "string",
+        title: "Project Name",
+      },
+      projectId: {
+        type: "string" as "string",
+        title: "Project Id",
+        value: "hi",
+      },
+      owner: {
+        type: "string" as "string",
+        title: "Owner",
+        default: owner,
+      },
+      repo: {
+        type: "string" as "string",
+        title: "Repo",
+        default: repo,
+      },
+    },
+  },
+  uiSchema: {
+    projectName: {
+      "ui:data-cy": "project-name-input",
+      "ui:showErrors": false,
+    },
+    projectId: {
+      "ui:data-cy": "project-id-input",
+      "ui:description":
+        "Optionally enter an immutable project ID that would be used by Evergreen internally instead of defaulting to a random hash; An id should only be user-specified with good reason, such as if the project will be using performance tooling. \n It cannot be changed!",
+    },
+    owner: {
+      "ui:data-cy": "owner-input",
+      "ui:showErrors": false,
+    },
+    repo: {
+      "ui:data-cy": "repo-input",
+      "ui:showErrors": false,
+    },
+  },
+});
+interface Props {
+  owner: string;
+  repo: string;
+}
+export const CreateProjectModal: React.FC<Props> = ({ owner, repo }) => {
+  const dispatchToast = useToastContext();
+  const modalFormDefinition = getModalFormDefinition(owner, repo);
+  const isAdmin = true; // todo after EVG-16353
+  const [open, setOpen] = useState(false);
+
+  const onCancel = () => setOpen(false);
+  const onConfirm = (formState) => {
+    createProject({
+      variables: {
+        project: {
+          identifier: formState.projectName,
+          id: formState.projectId,
+          owner: formState.owner,
+          repo: formState.repo,
+        },
+      },
+    });
+    setOpen(false);
+  };
+  const [formState, setFormState] = useState({});
```

Comment:
[nit] If you wanted you could initialize the form state with the owner and repo fields here. I think that may make more sense so that `getModalFormDefinition` could be a variable rather than a function, but up to you!

Reply:
not a bad idea. I think I'll keep it this way for now though because I would want a function to fetch the ModalFormDefinition since it's quite large so I want to keep it out of the component.